pkgs/development/haskell-modules/configuration-common.nix

```diff
@@ -3135,4 +3135,9 @@ self: super: {
     hash = "sha256-tFOWpjGmZANC7H82QapZ36raaNWuZ6F3BgjxnfTXpMs=";
   }) super.proto3-wire;
 
+  # Fix strictDeps build error "could not execute: hspec-discover"
+  safe-exceptions = super.safe-exceptions.overrideAttrs (prevAttrs: {
```

Comment:
`overrideAttrs` isn't used elsewhere in this file, instead `overrideCabal` is used instead.
Seems you need to add it to `testHaskellDepends` instead of `nativeBuildInputs` https://github.com/fpco/safe-exceptions/blob/master/test/Spec.hs